server/src/main.rs

```diff
@@ -0,0 +1,55 @@
+use clap::{App, Arg};
+use shadow_rs::shadow;
+
+pub mod management_api {
+    tonic::include_proto!("management");
+}
+pub mod attestation_api {
+    tonic::include_proto!("attestation");
+}
+
+#[macro_use]
+extern crate log;
+shadow!(build);
+
+mod attestation;
+mod management;
+mod server;
+mod users;
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+
+    let version = format!(
+        "\nv{}\ncommit: {}\nbuildtime: {}",
+        build::PKG_VERSION,
+        build::COMMIT_HASH,
+        build::BUILD_TIME
+    );
+
+    let matches = App::new("attestation-server")
+        .version(version.as_str())
+        .long_version(version.as_str())
+        .author("Confidential-Containers Team")
+        .arg(
+            Arg::with_name("listen")
+                .short("l")
+                .long("listen")
+                .value_name("sockaddr")
+                .help("Socket that the server will listen on to accept attestation & configuration requests.")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let sockaddr = match matches.is_present("listen") {
+        true => matches.value_of("listen").unwrap().to_string(),
+        false => "127.0.0.1:1234".to_string(),
+    };
+
+    debug!("Server listen addr: {}", &sockaddr);
+    let res = server::server(&sockaddr).await;
```

Comment:
So we're going to serve both the management and attestation api on the same port?

Reply:
Yes, I think it's better to keep the server's launch arguments as simple as possible. So a potential user needn't to distingush each port's functionalities. And for the potential security problem (because the managment API is sensitive), we can handle it like this:
- If the connection is built with a `normal` account, the server will only accept attestation request.
- If the connection is built with a `admin` account, the server will accept both the attestation and configuration requests.

Reply:
The attestation agent uses two different ports or the get resource and keyprovider APIs. I think that makes sense when the endpoints are used by different clients. I don't have a strong opinion here, but I think two might be better. It seems like the authentication for different accounts isn't implemented yet. If we use two different ports we could expose the attestation port to the network while keeping the other port hidden. I guess in the short term everything will be local anyway since the AS and KBS will be on the same node.

Reply:
Yes, It's reasonable to reach to better performance by hide the management port explictly. I will splict it to two ports.